tritium-metrics-jvm/src/main/java/com/palantir/tritium/metrics/jvm/JvmMetrics.java

```diff
@@ -132,48 +134,76 @@ private static void registerJvmMemory(TaggedMetricRegistry registry) {
             log.warn("Failed to load the MemoryMXBean, jvm.memory metrics will not be recorded");
             return;
         }
+        registerJvmMemory(registry, memoryBean);
+    }
+
+    @VisibleForTesting
+    static void registerJvmMemory(TaggedMetricRegistry registry, MemoryMXBean memoryBean) {
         JvmMemoryMetrics metrics = JvmMemoryMetrics.of(registry);
         // jvm.memory.total
-        metrics.totalInit((Gauge<Long>) () -> memoryBean.getHeapMemoryUsage().getInit()
-                + memoryBean.getNonHeapMemoryUsage().getInit());
-        metrics.totalUsed((Gauge<Long>) () -> memoryBean.getHeapMemoryUsage().getUsed()
-                + memoryBean.getNonHeapMemoryUsage().getUsed());
-        metrics.totalMax((Gauge<Long>) () -> memoryBean.getHeapMemoryUsage().getMax()
-                + memoryBean.getNonHeapMemoryUsage().getMax());
+        metrics.totalInit(
+                (NonNegativeGauge) () -> memoryBean.getHeapMemoryUsage().getInit()
+                        + memoryBean.getNonHeapMemoryUsage().getInit());
```

Comment:
are these computations problematic if one or both are not available?

Reply:
Probably worth checking both, in practice they’re all or nothing in my experience, but that could change.